### PR TITLE
[k8sTs] STAC-18524: Add --set-string 'logsAgent.enabled'='true' to k8s install

### PR DESCRIPTION
**Pull request by craffit**

### CODE REVIEW

File: setup/agent/k8sTs-kubernetes-openshift.md [hunk 1]
```diff
@@ -0,0 +1,582 @@
...
+* StackState Agent V2 on each node in the cluster
+* StackState Cluster Agent on one node
+* StackState Checks Agent on one node
+* kube-state-metrics
+
+To integrate with other services, a separate instance of StackState Agent V2 should be deployed on a standalone VM.
+
+# StackState Agent types
+
+The Kubernetes and OpenShift integrations collect topology data from Kubernetes and OpenShift clusters respectively, as well as metrics and events. To achieve this, different types of StackState Agent are used:
```
[the remark this answers is not in the document]
Comment (gggina) [reply setup/agent/k8sTs-kubernetes-openshift.md:22]: you can find the A-Z word list in the style guide here: https://learn.microsoft.com/en-us/style-guide/welcome/

File: setup/agent/k8sTs-kubernetes-openshift.md [hunk 2]
```diff
@@ -0,0 +1,582 @@
...
+
+Additional variables can be added to the standard helm command used to deploy the StackState Agent, Cluster Agent, Checks Agent and kube-state-metrics. For example:
+* It's recommended to [use a `stackstate.cluster.authToken`](#stackstateclusterauthtoken).
+* If you use a custom socket path, [set the `agent.containerRuntime.customSocketPath`](#agentcontainerruntimecustomsocketpath).
+
+{% hint style="info" %}
+Details of all available helm chart values can be found in the [Agent Helm Chart documentation \(github.com\)](https://github.com/StackVista/helm-charts/tree/master/stable/stackstate-agent).
+{% endhint %}
+
+### stackstate.cluster.authToken
```
[the remark this answers is not in the document]
Comment (craffit) [reply setup/agent/k8sTs-kubernetes-openshift.md:227]: keep

[the remark this answers is not in the document]
Comment (craffit) [reply setup/agent/k8sTs-kubernetes-openshift.md:227]: keep

File: setup/agent/k8sTs-kubernetes-openshift.md [hunk 3]
```diff
@@ -0,0 +1,582 @@
...
+
+It's recommended to use a `stackstate.cluster.authToken` in addition to the standard helm chart variables when the StackState Agent is deployed. This is an optional variable, however, if not provided a new, random value will be generated each time a helm upgrade is performed. This could leave some pods in the cluster with an incorrect configuration.
```
[the remark this answers is not in the document]
Comment (craffit) [reply setup/agent/k8sTs-kubernetes-openshift.md:229]: keep

File: setup/agent/k8sTs-kubernetes-openshift.md [hunk 4]
```diff
@@ -0,0 +1,582 @@
...
+  --set-string 'global.extraEnv.open.STS_LOG_PAYLOADS'='true' \
+  --set 'agent.logLevel'='debug' \
+  --set 'agent.scc.enabled'=true \
+  --set 'kube-state-metrics.securityContext.enabled'=false \
+  stackstate-agent stackstate/stackstate-agent
+```
+{% endtab %}
+{% endtabs %}
+
+### agent.containerRuntime.customSocketPath
```
[the remark this answers is not in the document]
Comment (craffit) [reply setup/agent/k8sTs-kubernetes-openshift.md:264]: keep

File: setup/agent/k8sTs-kubernetes-openshift.md [hunk 5]
```diff
@@ -0,0 +1,582 @@
...
+
+StackState Agent V2 is deployed as a DaemonSet with one instance **on each node** in the cluster:
+
+* Host information is retrieved from the Kubernetes or OpenShift API.
+* Container information is collected from the Docker daemon.
+* Metrics are retrieved from kubelet running on the node and also from kube-state-metrics if this is deployed on the same node.
+
+## Checks Agent
+
+The StackState Checks Agent is an additional StackState Agent V2 pod that will run the cluster checks that are configured on the [StackState Cluster Agent](#cluster-agent).
```
[the remark this answers is not in the document]
Comment (craffit) [reply setup/agent/k8sTs-kubernetes-openshift.md:46]: keep

File: setup/agent/k8sTs-kubernetes-openshift.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (craffit) [reply setup/agent/k8sTs-kubernetes-openshift.md]: fixed

File: setup/agent/k8sTs-kubernetes-openshift.md [hunk 6]
```diff
@@ -0,0 +1,582 @@
...
+Cluster checks configured here are run by the deployed [StackState Checks Agent](kubernetes-openshift.md#checks-agent) pod.
+
+# Setup
+
+## Supported Kubernetes versions
+
+StackState Agent v2.19.x is supported to monitor the following versions of Kubernetes or OpenShift:
+
+* Kubernetes:
+  * Kubernetes 1.16 - 1.24
```
[the remark this answers is not in the document]
Comment (gggina) [reply setup/agent/k8sTs-kubernetes-openshift.md:74]: keep

[the remark this answers is not in the document]
Comment (gggina) [reply setup/agent/k8sTs-kubernetes-openshift.md:74]: keep

File: setup/agent/k8sTs-kubernetes-openshift.md [hunk 7]
```diff
@@ -0,0 +1,582 @@
...
+  * EKS (with Kubernetes 1.16 - 1.24)
```
[the remark this answers is not in the document]
Comment (gggina) [reply setup/agent/k8sTs-kubernetes-openshift.md:75]: ignore

[the remark this answers is not in the document]
Comment (gggina) [reply setup/agent/k8sTs-kubernetes-openshift.md:75]: ignore

File: setup/agent/k8sTs-kubernetes-openshift.md [hunk 8]
```diff
@@ -0,0 +1,582 @@
...
+* OpenShift:
+  * OpenShift 4.3 - 4.11
```
[the remark this answers is not in the document]
Comment (gggina) [reply setup/agent/k8sTs-kubernetes-openshift.md:77]: ignore

File: setup/agent/k8sTs-kubernetes-openshift.md [hunk 9]
```diff
@@ -0,0 +1,582 @@
...
+* Default networking
+* Container runtime:
+  * Docker
+  * containerd
```
[the remark this answers is not in the document]
Comment (gggina) [reply setup/agent/k8sTs-kubernetes-openshift.md:81]: added to accept.txt

File: setup/agent/k8sTs-kubernetes-openshift.md [hunk 10]
```diff
@@ -0,0 +1,582 @@
...
+  * CRI-O
```
[the remark this answers is not in the document]
Comment (gggina) [reply setup/agent/k8sTs-kubernetes-openshift.md:82]: added to accept.txt

File: setup/agent/k8sTs-kubernetes-openshift.md [hunk 11]
```diff
@@ -0,0 +1,582 @@
...
+
+## Install
+
+The StackState Agent, Cluster Agent, Checks Agent and kube-state-metrics can be installed together using the StackState Agent Helm Chart:
+
+* [Online install](#online-install) - charts are retrieved from the default StackState chart repository (https://helm.stackstate.io), images are retrieved from the default StackState image registry (quay.io).
+* [Air gapped install](#air-gapped-install) - images are retrieved from a local system or registry.
```
[the remark this answers is not in the document]
Comment (gggina) [reply setup/agent/k8sTs-kubernetes-openshift.md:89]: Use Air-gapped in place of air gapped

File: setup/agent/k8sTs-kubernetes-openshift.md [hunk 12]
```diff
@@ -0,0 +1,582 @@
...
+    helm repo update
+   ```
+
+2. Deploy the StackState Agent, Cluster Agent, Checks Agent and kube-state-metrics to namespace `stackstate` using the helm command below.
+   * `<STACKSTATE_RECEIVER_API_KEY>` is set during StackState installation.
+   * `<STACKSTATE_RECEIVER_API_ADDRESS>` is specific to your installation of StackState.
+
+   * For details see [StackState Receiver API](/setup/agent/about-stackstate-agent.md#connect-to-stackstate).
+
+   - Note that [additional optional configuration](#helm-chart-values) can be added to the standard helm command.
```
[the remark this answers is not in the document]
Comment (gggina) [reply setup/agent/k8sTs-kubernetes-openshift.md:109]: keep

File: setup/agent/k8sTs-kubernetes-openshift.md [hunk 13]
```diff
@@ -0,0 +1,582 @@
...
+--set-string 'stackstate.cluster.name'='<OPENSHIFT_CLUSTER_NAME>' \
+--set-string 'stackstate.url'='<STACKSTATE_RECEIVER_API_ADDRESS>' \
+--set 'agent.scc.enabled'=true \
+--set 'kube-state-metrics.securityContext.enabled'=false \
+stackstate-agent stackstate/stackstate-agent
+```
+{% endtab %}
+{% endtabs %}
+
+### Air gapped install
```
[the remark this answers is not in the document]
Comment (gggina) [reply setup/agent/k8sTs-kubernetes-openshift.md:139]: fixed

**Pull request closed without merging**